File: GotlandBackend/Models/AllocationListItem.cs
using System;

namespace Accidis.Gotland.WebService.Models
{
	public sealed class AllocationListItem
	{
		public Guid CabinId { get; set; }
		public string Reference { get; set; }
		public string TeamName { get; set; }
		public string Note { get; set; }
		public int NumberOfPax { get; set; }
		public int TotalPax { get; set; }
	}
}

File: GotlandFrontend/lib/model/allocation_list_item.dart
import 'json_field.dart';

class AllocationListItem {
	final String cabinId;
	final String reference;
	final String teamName;
	final String note;
	final int numberOfPax;
	final int totalPax;

	AllocationListItem(this.cabinId, this.reference, this.teamName, this.note, this.numberOfPax, this.totalPax);

	factory AllocationListItem.fromMap(Map<String, dynamic> json) =>
		AllocationListItem(
			json[CABIN_ID],
			json[REFERENCE],
			json[TEAM_NAME],
			json[NOTE],
			json[NUMBER_OF_PAX],
			json[TOTAL_PAX],
		);
}
